src/components/ArticleCard.jsx

```diff
@@ -5,10 +5,60 @@ import CardContent from "@mui/joy/CardContent";
 import CardOverflow from "@mui/joy/CardOverflow";
 import Divider from "@mui/joy/Divider";
 import Typography from "@mui/joy/Typography";
-import ThumbUpOutlinedIcon from "@mui/icons-material/ThumbUpOutlined";
+import ThumbUpOffAltIcon from "@mui/icons-material/ThumbUpOffAlt";
+import ThumbDownOffAltIcon from "@mui/icons-material/ThumbDownOffAlt";
 import PersonOutlineOutlinedIcon from "@mui/icons-material/PersonOutlineOutlined";
+import { patchArticleVotes } from "../utils/api";
+import { useState } from "react";
+import Alert from "@mui/joy/Alert";
+
+const ArticleCard = ({ article, setArticles }) => {
+  const [isError, setIsError] = useState(false);
+
+  const handleUpvote = (selectedArticle) => {
+    setArticles((currentArticles) => {
+      return currentArticles.map((article) => {
+        if (article === selectedArticle) {
+          return { ...article, votes: article.votes + 1 };
+        }
+        return article;
+      });
+    });
+    patchArticleVotes(article.article_id, 1)
+      .then(() => {
+        setIsError(false);
+      })
+      .catch((err) => {
+        setIsError(true);
+      });
+  };
+
+  const handleDownvote = (selectedArticle) => {
+    setArticles((currentArticles) => {
+      return currentArticles.map((article) => {
+        if (article === selectedArticle) {
+          return { ...article, votes: article.votes - 1 };
+        }
+        return article;
+      });
+    });
+    patchArticleVotes(article.article_id, -1)
+      .then(() => {
+        setIsError(false);
+      })
+      .catch((err) => {
+        setIsError(true);
+      });
+  };
```

Comment:
handleUpvote and handleDownvote are very similar - you might be able to create on function that handles both directions of voting to keep your code DRY